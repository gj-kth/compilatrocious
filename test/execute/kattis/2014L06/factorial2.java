class Fac2 {
    public static void main(String[] args) {
        Factorial2 fac2;
        fac2 = new Factorial2();
        System.out.println(fac2.calc(0));
        System.out.println(fac2.calc(1));
        System.out.println(fac2.calc(7));
    }
}

class Factorial2 {
    public int calc(int num) {
        int ret;
        if (num < 1) {
            ret = 1;
        } else {
            ret = num*this.calc(num-1);
        }
        return ret;
    }
}