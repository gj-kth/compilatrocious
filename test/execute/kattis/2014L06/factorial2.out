1
1
5040
